runtime.md

```diff
@@ -15,3 +15,46 @@ Runs a process in a container. Can be invoked several times.
 Not sure we need that from oc cli. Process is killed from the outside.
 
 This event needs to be captured by oc to run onstop event handlers.
+
+## Hooks
+Hooks allow one to run code before/after various lifecycle events of the container. The state of the container is passed to the hooks over stdin, so the hooks could get the information they need to do their work.
+
+The path to the hooks could be absolute or relative and are executed from the host's filesystem. If they are relative, then they are searched for within the PATH environment variable on the host.
+
+The working directory for the hooks is set to the container's root filesystem unless overriden in the configuration.
+
+### Pre-start
+The pre-start hooks are called after the container process is started but before the user supplied command is executed. It provides an opportunity to customize the container. In Linux, for e.g., the network namespace could be configured in this hook.
+
+If a hook returns a non-zero exit code, then an error is returned to the caller and the container is torn down.
```

Comment:
Probably we should standardize this. Because it makes sense for implementation to know error text in addition to status code.